addon/registry.js

```diff
@@ -0,0 +1,57 @@
+let REGISTERED_RECEIVERS = {};
+
+export function registerReceiver(component) {
+  let componentName = getComponentRegistryKey(component);
+  REGISTERED_RECEIVERS[componentName] = component;
```

Comment:
Should this fail if there's already a registered receiver for the component name?

I'm imagining a scenario where you have something like:

```handlebars
{{random-color-component onClick=(action foo)}}

...

Does the second component overwrite the first here?
// Many moons later another engineer adds 
{{random-color-component onClick=(action bar)}}
```